wallet/handler_deprecated.go

```diff
@@ -0,0 +1,67 @@
+package wallet
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"math/big"
+
+	"github.com/golang/protobuf/proto"
+	typespb "github.com/vegaprotocol/api/grpc/clients/go/generated/code.vegaprotocol.io/vega/proto"
+)
+
+func (h *Handler) SignTx(token, tx, pubKey string, blockHeight uint64) (SignedBundle, error) {
+	h.mu.RLock()
+	defer h.mu.RUnlock()
+
+	// first the transaction would be in base64, let's decode
+	rawTx, err := base64.StdEncoding.DecodeString(tx)
+	if err != nil {
+		return SignedBundle{}, err
+	}
+
+	kp, err := h.getKeyPair(token, pubKey)
+	if err != nil {
+		return SignedBundle{}, err
+	}
+
+	if kp.Tainted {
+		return SignedBundle{}, ErrPubKeyIsTainted
+	}
+
+	txTy := &typespb.Transaction{
+		InputData:   rawTx,
+		Nonce:       makeNonce(),
+		BlockHeight: blockHeight,
+		From: &typespb.Transaction_PubKey{
+			PubKey: kp.pubBytes,
+		},
+	}
+
+	rawTxTy, err := proto.Marshal(txTy)
+	if err != nil {
+		return SignedBundle{}, err
+	}
+
+	// then lets sign the stuff and return it
+	sig, err := kp.Algorithm.Sign(kp.privBytes, rawTxTy)
+	if err != nil {
+		return SignedBundle{}, err
+	}
+
+	return SignedBundle{
+		Tx: rawTxTy,
+		Sig: Signature{
+			Sig:     sig,
+			Algo:    kp.Algorithm.Name(),
+			Version: kp.Algorithm.Version(),
+		},
+	}, nil
+}
+
+func makeNonce() uint64 {
+	max := &big.Int{}
+	// set it to the max value of the uint64
+	max.SetUint64(^uint64(0))
+	nonce, _ := rand.Int(rand.Reader, max)
+	return nonce.Uint64()
+}
```

Comment:
Didn't you add a NewNonce method somehwhere?

Reply:
🎯